src/main/scala/chisel3/util/experimental/package.scala

```diff
@@ -0,0 +1,34 @@
+package chisel3.util
+
+import chisel3.Module
+
+package object experimental {
+
+  /** Helper functions to emit module with implicit class
+    *
+    * @example
+    * {{{
+    *   import chisel3.util.experimental.ImplicitDriver
+    *   object Main extends App {
+    *     println((new Hello()).emitVerilog)
+    *   }
+    * }}}
+    */
+  implicit class ImplicitDriver(module: => Module) {
+    def emitVerilog = {
+      chisel3.stage.ChiselStage.emitVerilog(module)
+    }
+
+    def emitSystemVerilog = {
+      chisel3.stage.ChiselStage.emitSystemVerilog(module)
+    }
+
+    def emitFirrtl = {
+      chisel3.stage.ChiselStage.emitFirrtl(module)
+    }
+
+    def emitChirrtl = {
+      chisel3.stage.ChiselStage.emitChirrtl(module)
+    }
```

Comment:
We can bikeshed the names of these, but other ideas would be:

- `toVerilogString`
- `asVerilogString`

I would also suggest adding an entry-point for standard compilation:

```scala
  def compile(args: String *): AnnotationSeq = {
    (new ChiselStage).execute(args.toArray, Seq(ChiselGeneratorAnnotation(() => a)))
  }
```

Doing `(new Foo).compile("-X", "low")` is pretty slick.

Reply:
Got it!

Reply:
Re: bikeshed... I actually don't care. I think `emit*` is good and matches what we've used for a long time. No reason to change.

Reply:
How about API like this? I think we should also expose annotation interface to users:
```scala
    def execute(args: String*)(annos: AnnotationSeq = Nil): AnnotationSeq =
      (new ChiselStage)
        .execute(
          args.toArray,
          annos ++ Seq(new ChiselGeneratorAnnotation(() => module))
        )

    def compile(args: String*): AnnotationSeq = execute(args: _*)(Nil)
```
But which means executing is a little strange with 2 brackets.
```scala
    (new GCD).execute("xxx", "xxx")(Seq(SomeAnno0, SomeAnno1))
```

Reply:
> Re: bikeshed... I actually don't care. I think emit* is good and matches what we've used for a long time. No reason to change.

Personally I think `emit` sounds like a emission to file which has side effect, while `toXXX` has not. But I think both are OK if we need to align with our older APIs.

Reply:
I would like to keep ```emitVerilog()``` for generating the files. Then I would choose a different/better name for just returning it as a string. I think we need two different names anyway.

Reply:
how about `generateVerilogString`?

Reply:
maybe 
```
emitVerilog(path: os.Path = os.pwd): Unit
verilog(): String
```
sounds more simple?

Reply:
Why not keep the parameters as they have been for ```emitVerilog```. Keeping the possibility of setting other options as well. I also would not like to force the user to use this (new) os library.

Reply:
Personally, I like to have an active verb in the function name, since the function is going to perform a lot of work. `verilog` looks like a getter which imho should not be performing any substantial work.

Reply:
I like ```generateVerilogString``` as it is clear what it is. Maybe a bit lengthy. If we copy Java/Scala convention we could shorten it to ```toVerilogString```.

Reply:
something maybe
```
generateVerilogString(): String
generateVerilogFile(directory: os.Path): os.Path
```
?

Reply:
I think @ekiwi is reasonable that we need a long word to indicate this process is expensive ;p
I personally prefer a API with a specific name(Thanks to IDEA code completion)